### PR TITLE
Feature/add gormv2

### PR DESCRIPTION
**Pull request by aalexanderkevin**
Support gorm v2

### CODE REVIEW

File: txmanager.go [hunk 1]
```diff
@@ -4,23 +4,27 @@ import (
...
 type GormTxManager struct {
-	db	*gorm.DB
+	db   *gorm.DB
+	dbv2 *gormv2.DB
 }
 
 // StartTxManager create TxManager with db
-func StartTxManager(db *gorm.DB) TxManager {
-	return &GormTxManager{db: db}
+func StartTxManager(db *gorm.DB, dbv2 *gormv2.DB) TxManager {
```
Comment (hendra-shortlyst) [inline txmanager.go]: @aalexanderkevin how about creating new Gorm2TxManager implementation, and the constructor function can be: NewGormTxManager(db *gormv2.DB)?

File: txmanager.go [hunk 2]
```diff
@@ -14,17 +14,28 @@ type TxFn func(ctx context.Context) error
...
+}
+
+type GormV2TxManager struct {
+	db *gormv2.DB
 }
 
 // StartTxManager create TxManager with db
-func StartTxManager(db *gorm.DB, dbv2 *gormv2.DB) TxManager {
-	return &GormTxManager{db: db, dbv2: dbv2}
+func StartTxManager(db *gorm.DB) TxManager {
```
Comment (hendra-shortlyst) [inline txmanager.go:28]: @aalexanderkevin how about name the constructor to:

`func NewGormTxManager(db *gormv2.DB) *GormV2TxManager`

?

Comment (aalexanderkevin) [reply txmanager.go:28]: done on this commit ya https://github.com/shortlyst-ai/go-txmanager/pull/3/commits/37d24b9d098ad776266778955fbd8aa1491046b5
@hendra-shortlyst

File: txmanager.go [hunk 3]
```diff
@@ -14,15 +16,28 @@ type TxManager interface {
 	WithTransaction(ctx context.Context, txfn TxFn) error
 }
 
+type TxManagerGormV2 interface {
```
Comment (hendra-shortlyst) [inline txmanager.go]: @aalexanderkevin I think we don't need this TxManagerGormV2, the signature is the same with TxManager. Wdyt?

Comment (aalexanderkevin) [reply txmanager.go]: thanks mas done ya https://github.com/shortlyst-ai/go-txmanager/pull/3/commits/b95e120d6cd024e722de7cafffe5450a00e4aadf
@hendra-shortlyst

**Pull request merged**